src/local-time.lisp

```diff
@@ -211,7 +211,18 @@
      finally
         (return (if (zerop start) 0 (1- start)))))
 
-(defun %subzone-as-of (timezone seconds days)
+(defun %subzone-as-of (timezone seconds days &optional guess-p)
```

Comment:
I really don't like the call signature this produces with an unmarked boolean at the end, but introducing it as a `&key` argument will slow it down considerably. This is why I made `guess-offset` as a function instead of integrating it here. It also allowed for people to write their own offset-guessers.

Reply:
To my understanding, there is no longer 'guessing' involved. Except for the duration of offset-new - offset-old during a transition, the encoding is deterministic. And that period of ambiguity is not about guessing something that may or may not be right, but there are simply zero or two correct answers.

My suggestion: the fourth argument is required and renamed to encode-p or wall-time-p.

Or do you have a use case where guessing is required?

For the period with no correct answer (02:30 on a day when the clock is moved from 02:00 to 03:00) should there be an additional strict mode?
For the period with two correct answers, should there be a warning, or a way of chosing? When the clock is moved back from 02:00 CEST to 01:00 CET, I think 01:30 should best be regarded as CET. You might argue that it depends on the current time (it is CEST while in CEST and CET while in CET);you can have options:
- later (report 01:30 CET), with variant later-warn (no warning / warning)
- earlier (report 01:30 CEST) with variant earlier-warn (no warning / warning)
- error : refuse to encode
- nearest : report 01:30 CEST when actual time is before the transition and 01:30 CET when actual time is after

If I had time, I would check how other libraries do this, but alas.

To keep the 'write their own offset-guessers', %subzone-as-of might return more than one subzone when wall-time-p is true and an intermediate function (which might be redefine) would handle the different cases.

Reply:
Hi Daniel, I'd like to go on with this and the other changes. Tell me if you want this completely reverted, or maybe just keep a user-definable guess-offset with the current semantics?

Reply:
Perfectly okay! Do I understand correctly you will merge on the main repo, or should I make a branch on my fork?

Reply:
I've edited the PR to merge into the branch pr-116. I can merge that and we can continue there.

Reply:
oooor I can screw that up and yolo it straight into master.

Reply:
Well, that liberates us from the perfect commit history syndrome. Thanks!